pkg/action/install.go

```diff
@@ -152,22 +157,46 @@ func (i *Install) InstallAllSharedDeps(chrt *chart.Chart, settings *cli.EnvSetti
 		prefix = fmt.Sprintf("%*s", lvl*2, "- ")
 	}
 
-	for _, dep := range deps {
+	for _, dep := range sharedDeps {
+		switch i.OptionalDeps {
+		case "all":
+			// install all deps, optional or not
+		case "none":
+			if dep.IsOptional {
+				continue
+			}
+		case "ask":
+			if dep.IsOptional {
+				if !promptUserForOptionalDependencyInstall(dep, logger, os.Stdin) {
+					continue
+				}
+			}
+		default:
+			return errors.New("Incorrect value for --optional-deps. Valid values: [default=ask|all|none]")
```

Comment:
Error output with flags shouldn't be here. We should return an error fit for the library, and catch it on cmd.